.gitlab-ci.yml

```diff
@@ -325,6 +325,23 @@ pkg:nix:deploy:
     - master
     - /^v.*\..*$/
 
+pkg:nix:deploy:channel:
+  stage: deploy
+  environment:
+    name: cachix
+    url: https://coq.cachix.org
+  dependencies:
+    - pkg:nix:deploy
+  before_script:
+    - which ssh-agent || ( apt-get update -y && apt-get install openssh-client -y )
+    - eval $(ssh-agent -s)
+    - echo "$GITHUB_DEPLOY_KEY" | tr -d '\r' | ssh-add - > /dev/null
+    - mkdir -p ~/.ssh
+    - chmod 700 ~/.ssh
+    - ssh-keyscan -t rsa github.com >> ~/.ssh/known_hosts
+  script:
+    - git push git@github.com:coq/coq "${CI_COMMIT_REF_NAME}-cachix"
```

Comment:
What happens in case of a race? E.g., if an earlier job finishes after a more recent one?

Reply:
Good point. It would fail but there is probably a way to tell git to exit with success if the pushed commit is already in the remote...